lampo-common/src/conf.rs

```diff
@@ -180,26 +180,26 @@ impl TryFrom<String> for LampoConf {
         // Strip the value of whitespace
         let node = node.to_trimmed();
 
-        let mut core_url = None;
-        let mut core_user = None;
-        let mut core_pass = None;
-        if node == "core" {
-            core_url = conf
-                .get_conf("core-url")
-                .map_err(|err| anyhow::anyhow!("{err}"))?;
-            // If the value isn't none, strip the value of whitespace
-            core_url = core_url.map(|url| url.to_trimmed());
+        let Some(core_url) = conf
+            .get_conf("core-url")
+            .map_err(|err| anyhow::anyhow!("{err}"))?
+        else {
+            anyhow::bail!("Core URL must be specified")
+        };
```

Comment:
This look wired, you are already returning an error with `map_err()`? why you are not writing it in this way

```suggestion
        let Some(core_url) = conf
            .get_conf("core-url")
            // if there is a error working with the config file internal
            .map_err(|err| anyhow::anyhow!("{err}"))?
            // if there is no value inside the configuration file
            .ok_or("Core URL must be specified")?;
```

it is much more easy to read IMHO and understand what it is going on